translator/navjoy_translator.py

```diff
@@ -0,0 +1,347 @@
+            coordinates = polygon_tools.polygon_to_polyline_center(polygon)
+        else:
+            logging.warning(
+                f"Invalid event with id = {obj.get('sys_gUid')}. No polygon or linestring")
+            return None
+
+    if not coordinates:
+        logging.warning(
+            f"Invalid event with id = {obj.get('sys_gUid')}. No valid coordinates found")
+        return False
+
+    # [0].get('coordinates')[0]
+    # coordinates = polygon_tools.polygon_to_polyline_center(polygon)
+
+    # Reverese polygon if it is in the opposide direction as the message
+    polyline_direction = polygon_tools.get_road_direction_from_coordinates(
+        coordinates)
+    if direction == REVERSED_DIRECTION_MAP.get(polyline_direction):
+        coordinates.reverse()
+
+    geometry = {}
+    if len(coordinates) > 2:
+        geometry['type'] = "LineString"
+    else:
+        geometry['type'] = "MultiPoint"
+
+    geometry['coordinates'] = coordinates
+    properties = wzdx_translator.initialize_feature_properties()
+
+    # Event Type ['work-zone', 'detour']
+    properties['event_type'] = 'work-zone'
+
+    # start_date
+    start_date = date_tools.parse_datetime_from_iso_string(
+        data.get('workStartDate'))
+    properties['start_date'] = date_tools.get_iso_string_from_datetime(
+        start_date)
+
+    # end_date
+    end_date = date_tools.parse_datetime_from_iso_string(
+        data.get('workEndDate'))
+    if end_date:
+        properties['end_date'] = date_tools.get_iso_string_from_datetime(
+            end_date)
+    else:
+        properties['end_date'] = ''
+
+    # start_date_accuracy
+    properties['start_date_accuracy'] = "estimated"
+
+    # end_date_accuracy
+    properties['end_date_accuracy'] = "estimated"
+
+    # beginning_accuracy
+    properties['beginning_accuracy'] = "estimated"
+
+    # ending_accuracy
+    properties['ending_accuracy'] = "estimated"
+
+    # road_name
+    properties['road_names'] = [data.get('streetNameFrom')]
+
+    # direction
+    properties['direction'] = direction
+
+    # vehicle impact
+    properties['vehicle_impact'] = get_vehicle_impact(
+        data.get('reductionJustification'))
+
+    # event status
+    properties['event_status'] = date_tools.get_event_status(
+        start_date, end_date)
+
+    # type_of_work
+    # maintenance, minor-road-defect-repair, roadside-work, overhead-work, below-road-work, barrier-work, surface-work, painting, roadway-relocation, roadway-creation
+    projectDescription = data.get('descriptionForProject')
+    types_of_work = get_types_of_work(projectDescription)
+    if types_of_work:
+        properties['types_of_work'] = types_of_work
+
+    # reduced_speed_limit
+    reducedSpeed = data.get('requestedTemporarySpeed')
+    if reducedSpeed:
+        properties['reduced_speed_limit'] = wzdx_translator.string_to_number(
+            reducedSpeed)
+
+    # description
+    reductionJustification = data.get('reductionJustification')
```

Comment:
The only reason we would have a reduction justification is in conjunction with a reduced speed right? Could we put this block under the `if reducedSpeed` above?

Reply:
I went and removed the if statement, it was not doing anything. We know that there should be a reduced speed limit, otherwise the message is useless